backend/corpora/parliament/utils/parlamint_v4.py

```diff
@@ -0,0 +1,198 @@
+from addcorpus.python_corpora.extract import XML, Combined, Metadata
+from bs4.element import NavigableString
+import datetime
+
+"""
+This file was created as an updated utils file for the ParlaMint dataset, version 4.0. The previous utils file
+is based on version 2.0.
+"""
+
+POLITICAL_ORIENTATIONS = {
+    'L': 'Left',
+    'C': 'Centre',
+    'R': 'Right',
+    'FL': 'Far Left',
+    'FR': 'Far Right',
+    'CL': 'Centre Left',
+    'CR': 'Centre Right',
+    'CCL': 'Centre to Centre Left',
+    'CCR': 'Centre to Centre Right',
+    'CLL': 'Centre Left to Left',
+    'CRR': 'Centre Right to Right',
+    'LLF': 'Left to Far Left',
+    'RRF': 'Right to Far Right',
+    'BT': 'Big Tent',
+    'NP': 'Nonpartisan',
+    'PP': 'Pirate Party',
+    'SI': 'Single Issue',
+    'SY': 'Syncretic',
+    'NA': ''
+}
+
+def clean_value(value):
+    if type(value) == str or type(value) == NavigableString:
+        stripped = value.strip()
+        if len(stripped):
+            return stripped
+    if type(value) == int or type(value) == float:
+        return value
+    return value
+
+def extract_org_data(node):
+    id = node['xml:id']
+
+    full_name_node = node.find('orgName', full='yes')
+    full_name = full_name_node.text if full_name_node else None
+
+    abbreviation_node = node.find('orgName', full='abb')
+    name = abbreviation_node.text if abbreviation_node else full_name or id
+
+    wikimedia_uri_node = node.find('idno', type='URI', subtype='wikimedia')
+    wikimedia_uri = wikimedia_uri_node.text if wikimedia_uri_node else None
+
+    political_orientation_node = node.find('state', type='politicalOrientation')
+    political_orientation = political_orientation_node.find('state')['ana'] if political_orientation_node else None
+
+    return {
+        'name': name,
+        'full_name': full_name,
+        'org_role': node['role'],
+        'id': id,
+        'wikimedia': wikimedia_uri,
+        'political_orientation': political_orientation
+    }
+
+def extract_all_org_data(soup):
+    orgs_list = soup.find('listOrg')
+    org_data = map(extract_org_data, orgs_list.find_all('org'))
+    make_id = lambda name: '#' + name
+    org_dict = {}
+    for org in org_data:
+        org_dict[make_id(org['id'])] = org
+    return org_dict
+
+def extract_person_data(node):
+    id = '#' + node['xml:id']
+    surname = node.persName.surname.text.strip()
+    forename = node.persName.forename.text.strip()
+    name = ' '.join([forename, surname])
+    role = node.persName.roleName.text.strip() if node.persName.roleName else None #too simple, needs to be able to have different values and be gotten from the org_node
```

Comment:
This is currently handled in the specific corpus defition, by a way that requires all org_nodes to be saved to the person, see my other comment.